### ISSUE TITLE
Support embedding FlutterView in an Android Fragment

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#17875]
Android fragments have a different lifecycle than activities. It would be common for existing apps to add FlutterView to their fragments and expect all the fragment transactions to work properly. Just like we have an `FlutterActivityDelegate`, we likely need a `FlutterFragmentDelegate` that wires in all the lifecycle events to FlutterView.

In a recent hackathon with an internal team, we discovered that the app hangs while transitioning out of a fragment containing a full screen flutter view to a fragment that contains native views. While this could not be reproduced in a simple app, the problem was resolved by calling `view.destroy()` and `view.detach()` before the fragment was removed via FragmentManager. Developers should not need to discover this on their own.

**Comment by moyaying** [comment]
I want 'FlutterFragment' also...

**Comment by mit-mit** [comment]
This has landed :-)

**Comment by mit-mit** [comment]
See this: https://github.com/flutter/flutter/wiki/Add-Flutter-to-existing-apps
